controller/space.go

```diff
@@ -66,18 +84,38 @@ func (c *SpaceController) Create(ctx *app.CreateSpaceContext) error {
 		*/
 
 		newArea := area.Area{
-			ID:      satoriuuid.NewV4(),
-			SpaceID: space.ID,
-			Name:    space.Name,
+			ID:      uuid.NewV4(),
+			SpaceID: rSpace.ID,
+			Name:    rSpace.Name,
 		}
 		err = appl.Areas().Create(ctx, &newArea)
 		if err != nil {
-			return jsonapi.JSONErrorResponse(ctx, errs.Wrapf(err, "failed to create area: %s", space.Name))
+			return jsonapi.JSONErrorResponse(ctx, errs.Wrapf(err, "failed to create area: %s", rSpace.Name))
 		}
 
 		res := &app.SpaceSingle{
-			Data: ConvertSpace(ctx.RequestData, space),
+			Data: ConvertSpace(ctx.RequestData, rSpace),
+		}
+
+		// Create keycloak resource for this space
+		resource, err := c.resourceManager.CreateResource(ctx, ctx.RequestData, rSpace.ID.String(), spaceResourceType, &rSpace.Name, &scopes, rSpace.OwnerId.String(), rSpace.Name+"-"+uuid.NewV4().String())
```

Comment:
actually intentionally added KC resource creation to the transaction. If
we failed to create a KC resource we should fail to create the space. But I
guess it could hit the DB performance or some DB limits if hold too many
transactions because of network slowness or something? Should we create a
new space then complete the transaction then create a new KC resource and
if it fails then delete the created space and fire an error instead?